hack/test/unit

```diff
@@ -18,16 +18,32 @@ TESTDIRS="${TESTDIRS:-./...}"
 exclude_paths='/vendor/|/integration'
 pkg_list=$(go list $TESTDIRS | grep -vE "($exclude_paths)")
 
-echo "${pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
-	&& if ! type docker-proxy; then
-		hack/make.sh binary-proxy install-proxy
-	fi
+base_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings -v "/libnetwork" || :)
+libnetwork_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings "/libnetwork" || :)
+
+echo "${libnetwork_pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
+    && if ! type docker-proxy; then
+        hack/make.sh binary-proxy install-proxy
+    fi
 
 mkdir -p bundles
-gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report.json --junitfile=bundles/junit-report.xml -- \
-	"${BUILDFLAGS[@]}" \
-	-cover \
-	-coverprofile=bundles/profile.out \
-	-covermode=atomic \
-	${TESTFLAGS} \
-	${pkg_list}
+
+if [ "${base_pkg_list}" ]; then
+    gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report.json --junitfile=bundles/junit-report.xml -- \
+    	"${BUILDFLAGS[@]}" \
+    	-cover \
+    	-coverprofile=bundles/profile.out \
+    	-covermode=atomic \
+    	${TESTFLAGS} \
+    	${base_pkg_list}
+fi
+if [ "${libnetwork_pkg_list}" ]; then
+    gotestsum --format=standard-quiet --jsonfile=bundles/libnetwork-go-test-report.json --junitfile=bundles/libnetwork-junit-report.xml -- \
```

Comment:
May need to check if the Jenkinsfile uploads the new files as well

Reply:
Don't have much experience with Jenkins - the existing functionality related to this is not trivial, I will have to study the docs, but any pointers meanwhile would be very welcome!

Reply:
I'm not great at Jenkinsfiles either (mostly "do by example, and a lot of Google search") 🤣

The essential bit is that each of the stages will have a step that uploads the results; we probably need to either add another one of those (for stages that run the libnetwork tests), or perhaps make it use a wildcard; see, e.g. https://github.com/moby/moby/blob/master/Jenkinsfile#L202

Reply:
I tried something, please confirm that it makes sense, since I am not sure how to test this.